### DISCUSSION TITLE
Zipcode is Invalid while updating Shipping Profile

### DISCUSSION THREAD

**Discussion by uzairfiaz-gelato** [etsy/open-api#664] [Q&A] [answered]
We are getting errors when updating existing Shipping profiles.
The used endpoint is: /v3/application/shops/{shop_id}/shipping-profiles/{shipping_profile_id}. 
In the response, we are getting this error `ZIP Code  is invalid`. The shipping profile had empty/null `origin_postal_code` set on time of creation. The `origin_country_iso` is NO and according to [documentation](https://developer.etsy.com/documentation/reference#operation/createShopShippingProfile) we do not need to set value for this.  

Can someone help us to understand why this problem occurs?

**Comment by etsyachristensen** [comment]
@uzairfiaz-gelato We will need a copy of the payload you are sending and the shipping_profile_id you are attempting to update. This will help us dig deeper to find out what the problem is. There are a few countries that do not require a zip code. That said, if you have the `origin_postal_code` input parameter in the payload, but leave it blank it will throw an error. If the country doesn't require the zip code, then leave the parameter out if you don't have a zip code to send. If you do have a zip code and send it, whether the country requires it or not, we will validate that the zip code is correct.

**Reply by uzairfiaz-gelato** [reply]
@etsyachristensen please find the attached payloads for create and update requests. I simply created a Shipping profile without `origin_postal_code` as suggested by you above and tried to update the same shipping profile and got the ZipCode Invalid error.

[Requests.zip](https://github.com/etsy/open-api/files/10038491/Requests.zip)

**Reply by etsyachristensen** [reply]
Thanks @uzairfiaz-gelato. We will do some digging.

**Comment by etsyachristensen** [comment]
Ok so the core fulfillment team actually made a change some time back that actually requires zip codes for most countries. The documentation on OpenAPI unfortunately did not get updated. We will correct this ASAP. Here is the list of country ISOs that require zip codes (so most of them):

AD, AL, AM, AR, AS, AT, AU, AX, AZ, BA, BB, BD, BE, BG, BH, BL, BM, BN, BR, BT, BY, CA, CC, CH, CL, CN, CO, CR, CV, CX, CY, CZ, DE, DK, DO, DZ, EC, EE, EG, EH, ES, ET, FI, FK, FM, FO, FR, GB, GE, GF, GG, GI, GL, GN, GP, GR, GS, GT, GU, GW, HM, HN, HR, HT, HU, ID, IE, IL, IM, IN, IO, IQ, IR, IS, IT, JE, JO, JP, KE, KG, KH, KR, KW, KY, KZ, LA, LB, LI, LK, LR, LS, LT, LU, LV, MA, MC, MD, ME, MF, MG, MH, MK, MM, MN, MP, MQ, MT, MU, MV, MX, MY, NC, NE, NF, NG, NI, NL, NO, NP, NZ, OM, PE, PF, PG, PH, PK, PL, PM, PR, PT, PW, PY, RE, RO, RS, RU, SA, SE, SG, SH, SI, SJ, SK, SM, SN, SO, SV, SZ, TC, TH, TJ, TM, TN, TR, TW, TZ, UA, UM, US, UY, UZ, VA, VC, VE, VG, VI, VN, WF, YT, ZA, ZM, ZZ

**Reply by uzairfiaz-gelato** [reply]
Thanks @etsyachristensen for quick replies, But unfortunately we sometimes get  `We currently do not support submitting postal codes for this country: NO` error for some stores while creating shipping profiles. So, we do not send `origin_postal_code` and then we get ZipCode Invalid error on updating the shipping profile. But it is interesting that we do not get any error with empty `origin_postal_code` when creating a Shipping profile.

**Shipping Profile Create Request:**
`{"title":"Gelato: Test","origin_country_iso":"NO","primary_cost":20,"secondary_cost":10,"min_processing_time":2,"max_processing_time":5,"destination_country_iso":"NO","destination_region":null,"origin_postal_code":"0191","min_delivery_days":1,"max_delivery_days":6}`

**Response:**
`{"errors":true,"status":400,"response":{},"body":{"error":"We currently do not support submitting postal codes for this country: NO","error_description":""},"bodyArray":{"error":"We currently do not support submitting postal codes for this country: NO","error_description":""},"link":null,"exception":{}}`

**Reply by etsyachristensen** [reply]
Yeah. I apologize. I got ahead of myself and forgot that the rule is that those zip codes are required if ORIGINATING from the US. We have a small bug in the system on the UPDATE method, which is why the CREATE works fine. We are working on making the adjustment.

**Reply by uzairfiaz-gelato** [reply]
Thanks @etsyachristensen. Do we have an ETA on when this will be fixed? As this is affecting a lot our customers.

**Reply by etsyachristensen** [reply]
Unfortunately I don't have an estimate for you at this time.

**Reply by DemiImp** [reply]
I haven't double-checked all of those countries, but LK (Sri Lanka) does not appear to have zip codes.  A customer with a LK address did not have zip code on their account.  You can check Google Maps and addresses don't have zips.

`{"count":1,"results":[{"user_address_id":943227051219,"user_id":[redacted],"name":"MASTER CARD","first_line":"[redacted],","second_line":"","city":"devinuwara.","state":"","zip":"","iso_country_code":"LK","country_name":"Sri Lanka","is_default_shipping_address":true}]}`

@etsyachristensen - Are you sure that list is accurate?

**Comment by etsyachristensen** [comment, marked as answer]
@uzairfiaz-gelato We have just deployed a fix for this issue. I apologize that it took so long to get this into production.

**Reply by uzairfiaz-gelato** [reply]
Thanks, @etsyachristensen for updating. Can you please let me know what should be expected when sending empty `''` or `null` for `origin_postal_code`?
